pkg/kube/inject/webhook.go

```diff
@@ -808,7 +808,12 @@ func (wh *Webhook) serveInject(w http.ResponseWriter, r *http.Request) {
 	}
 
 	var reviewResponse *kubeApiAdmissionv1beta1.AdmissionResponse
-	ar := kubeApiAdmissionv1beta1.AdmissionReview{}
+	ar := kubeApiAdmissionv1beta1.AdmissionReview{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "admission.k8s.io/v1beta1",
```

Comment:
I am a bit confused how this works - maybe I don't understand what k8s is doing. My understanding is you are telling k8s to request v1, but then we are serving v1beta1.. don't we need to respond with what it requests?

What I would love to see is a test that sends two requests, one with v1beta1 and one with v1 request types, and verifies things work. Can probably be a pretty simple unit test.

Reply:
No. Right now we are working with `v1beta1`. This change only serves as disambiguation.

We are just paving the path for a `v1` transition.

Reply:
I have a feeling these go changes are causing the failures. I can revert them.

This kubernetes behaviour is rather inconsistent. hmmmm

Reply:
my understanding is the admissionReviewVersions will send a request using the first version supported by the server. Most versions 1.15+ support both and therefore would send a v1 request?

Note this is based in me reading the API doc, I may have misunderstood it

Reply:
So If I mention the apiVersion as `v1beta1` explicitly, wouldn't that be sent?

Reply:
what I verified - 1⃣️ installing with `v1beta1` (the one after the revert https://github.com/istio/istio/pull/26285) first, then 2⃣️ install with `v1` (before the revert), the I got this error.

```
$ ./istioctl-18v1 install  --set hub=gcr.io/istio-testing --set tag=1.8-alpha.601b09b15f29272d22c7dcc676f26dd120c9583e
✔ Istio core installed
2020-08-10T05:19:36.631481Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-10T05:19:36.796396Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-10T05:19:36.986405Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
2020-08-10T05:19:37.168168Z     error   installer       Internal error occurred: failed calling webhook "validation.istio.io": expected webhook response of admission.k8s.io/v1, Kind=AdmissionReview, got /, Kind=
```

Then I cleaned up istio, replayed steps 1⃣️ and 2⃣️. But in 1⃣️ I added `admissionReviewVersions: ["v1", "v1beta1"]` to webhook CRs. Then the error disappeared.